grails-app/domain/org/pih/warehouse/inventory/CycleCountRequest.groovy

```diff
@@ -0,0 +1,61 @@
+package org.pih.warehouse.inventory
+
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.core.User
+import org.pih.warehouse.product.Product
+
+class CycleCountRequest {
+
+    String id
+
+    Location facility
+
+    Product product
+
+    CycleCountRequestStatus status
+
+    // To be included when cycle count table is created
+    // CycleCount cycleCount
+
+    CycleCountRequestType requestType
+
+    Boolean blindCount
+
+    Date dateCreated
+
+    Date lastUpdated
+
+    User createdBy
+
+    User updatedBy
+
+    def beforeInsert() {
+        createdBy = AuthService.currentUser
+        updatedBy = AuthService.currentUser
+    }
+
+    def beforeUpdate() {
+        updatedBy = AuthService.currentUser
+    }
+
+    static constraints = {
+        id generator: "uuid"
+        product(nullable: true, unique: "facility")
```

Comment:
Is the cycle count request meant to be deleted at the end once it is completed? Right now if now if it won't be deleted, then it will prevent to requesting it again in the future due to this constraint. Was there a case when the product is null on the CCR level?